src/Raven.Server/Documents/Queries/AST/PatternMatchExpressEdge.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Sparrow;
+
+namespace Raven.Server.Documents.Queries.AST
+{
+    public class PatternMatchExpressEdge : PatternMatchExpression
```

Comment:
This should be removed. There is no such thing.
Instead, this should be a `with edge` expression, and the filter should be in the `where` clause

Reply:
Will inline the data about the edge in PatternMatchElementExpression

Reply:
What do you mean, inline?

Reply:
I will move the information about the edge (edge alias/edge type) to the PatternMatchElementExpression because it represents "to/from" pair on the pattern expression